substrate/client/rpc-spec-v2/src/chain_head/tests.rs

```diff
@@ -3297,3 +3330,120 @@ async fn storage_closest_merkle_value() {
 		merkle_values_rhs.get(&hex_string(b":AAAA")).unwrap()
 	);
 }
+
+#[tokio::test]
+async fn chain_head_single_connection_context() {
+	let server_addr = run_server().await;
+	let server_url = format!("ws://{}", server_addr);
+	let client = jsonrpsee::ws_client::WsClientBuilder::default()
+		.build(&server_url)
+		.await
+		.unwrap();
+	// Calls cannot be made from a different connection context.
+	let second_client = jsonrpsee::ws_client::WsClientBuilder::default()
+		.build(&server_url)
+		.await
+		.unwrap();
+
+	let mut sub: RpcClientSubscription<FollowEvent<String>> = client
+		.subscribe("chainHead_unstable_follow", rpc_params![true], "chainHead_unstable_unfollow")
```

Comment:
pretty sure that you could use the type API that is generated by the jsonrpsee proc macro here when you are using a "real ws client" (not sure about name though)

```rust
client.chainhead_unstable_follow(true)
```

Reply:
I completely forgot about those! I've added them since they improve a lot the code readability, thanks!